extensions/applicationinsights-clickanalytics-js/src/DataCollector.ts

```diff
@@ -115,15 +115,8 @@ export function getPageName(config: IClickAnalyticsConfiguration, overrideValues
     } else if (config.coreData && config.coreData.pageName) {
         return config.coreData.pageName;
     } else {
-        let loc = getLocation() || ({} as Location);
-        var pagename = loc.pathname || "";
```

Comment:
This feels like we should have a config for this, to allow users to determine whether they use the document title vs the URL. 

This is especially true nowadays that most browsers don't show or minimize the document title -- so it may not always be present.

Reply:
I was thinking users can override the pageName using the config.coreData.pageName config.

Reply:
We provide a callback [config](https://github.com/microsoft/ApplicationInsights-JS/blob/9271cf7bf59495e0d696688054d29b14316fc098/extensions/applicationinsights-clickanalytics-js/src/Interfaces/Datamodel.ts#L116) to override the default pageName capturing behavior.